### PR TITLE
netty: support `status()` on Headers

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
Recent Netty change https://github.com/netty/netty/commit/a91df58ca17d5b30c57c46dde5b1d60bb659b029
caused the `status()` method to be invoked, which AbstractHttp2Headers does not implement.
This change is necesary to upgrade to Netty 4.1.14

### CODE REVIEW

File: netty/src/main/java/io/grpc/netty/GrpcHttp2OutboundHeaders.java [hunk 1]
```diff
@@ -67,6 +67,16 @@ private GrpcHttp2OutboundHeaders(AsciiString[] preHeaders, byte[][] serializedMe
   }
 
   @Override
+  @SuppressWarnings("ReferenceEquality") // STATUS.value() never changes.
+  public CharSequence status() {
+    // preHeaders is never null.  It has status as the first element or not at all.
+    if (preHeaders.length >= 1 && preHeaders[0] == Http2Headers.PseudoHeaderName.STATUS.value()) {
+      return Http2Headers.PseudoHeaderName.STATUS.value();
```
Comment (ejona86) [inline netty/src/main/java/io/grpc/netty/GrpcHttp2OutboundHeaders.java]: Shouldn't this be `preHeaders[1]`?

Comment (carl-mastrangelo) [inline netty/src/main/java/io/grpc/netty/GrpcHttp2OutboundHeaders.java]: Interestingly yes, but it doesn't make the test fail.  The reason this method is called is to check for 1XX status codes, which we don't use.

File: netty/src/main/java/io/grpc/netty/GrpcHttp2OutboundHeaders.java [hunk 2]
```diff
@@ -67,6 +67,16 @@ private GrpcHttp2OutboundHeaders(AsciiString[] preHeaders, byte[][] serializedMe
   }
 
   @Override
+  @SuppressWarnings("ReferenceEquality") // STATUS.value() never changes.
+  public CharSequence status() {
+    // preHeaders is never null.  It has status as the first element or not at all.
+    if (preHeaders.length >= 1 && preHeaders[0] == Http2Headers.PseudoHeaderName.STATUS.value()) {
+      return Utils.STATUS_OK;
```
Comment (ejona86) [inline netty/src/main/java/io/grpc/netty/GrpcHttp2OutboundHeaders.java]: Please don't hard-code it to this degree.

Comment (carl-mastrangelo) [inline netty/src/main/java/io/grpc/netty/GrpcHttp2OutboundHeaders.java]: Done.

**Pull request merged**